components/builder-protocol/src/originsrv.rs

```diff
@@ -542,6 +543,37 @@ impl FromStr for OriginPackageIdent {
     }
 }
 
+impl FromStr for OriginPackageIdentWithChannels {
```

Comment:
It has a `channels` property. 😄  

Originally I considered adding a channels property to `OriginPackageIdent`, but in talking that over with @chefsalim last week, it seems `OriginPackageIdent` is more of a fundamental structure, and that channels really only make sense in a subset of situations. So we chose to make a new definition just for structures with channels.